src/MacVim/MMWindowController.m

```diff
@@ -1567,6 +1567,12 @@ - (void)updateToolbar
 
 - (BOOL)maximizeWindow:(int)options
 {
+    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_10_Max) {
```

Comment:
This basically fixes the problem for me, running on El Capitan. Thanks!

I'm wondering if this conditional should also check to see if the window is actually in split mode.

So far, everything I've tried has worked, even old-style full-screen mode. But I'm wondering if some non-split-screen call to maximizeWindow really does want the code below.